### DISCUSSION TITLE
Switching the default target domain for doors and windows from cover to binary_sensor

### DISCUSSION THREAD

**Discussion by tetele** [OHF-Voice/intents#2168] [_Polls]
At the moment, only a single `domain` can be used for querying entities. Unfortunately, doors and windows are represented either as entities with `{domain: 'cover', device_class: 'door'}` or `{domain: 'binary_sensor', device_class: 'door'}` (and their `window` equivalents).

At some point, we decided that the `cover` domain should trump the `binary_sensor` domain in sentences such as `How many doors are open?` and, as such, people who have `binary_sensor` doors get a false response. That's not a problem, it's a design decision that was made.

However, judging by my own experience, support questions and available devices, I feel as though `cover` doors and windows are **FAR** less prevalent than door/window sensors. I was wondering if it would not be a good idea to change the sentences to point to the `binary_sensor` domain by default.

**Comment by davefx** [comment]
I honestly think that the problem is that we have binary_sensors vs covers.
A door or window is a cover, no matter if it's actionable or not. IMHO,
Home Assistant should allow the easy creation of non-actionable covers from
binary_sensors.

--
David Marín Carreño ***@***.***>

**Reply by tetele** [reply]
Good idea! I've made an [architectural proposal](https://github.com/home-assistant/architecture/discussions/1084) for this.

**Comment by tetele** [comment]
Alternative solution proposed here https://github.com/home-assistant/architecture/discussions/1084
